addons/mrp/models/mrp_production.py

```diff
@@ -289,24 +290,19 @@ def _compute_mrp_production_backorder(self):
         for production in self:
             production.mrp_production_backorder_count = len(production.procurement_group_id.mrp_production_ids)
 
-    @api.depends('move_raw_ids', 'state')
+    @api.depends('move_raw_ids', 'state', 'date_planned_start')
     def _compute_components_availability(self):
         self.components_availability = False
         self.components_availability_state = 'available'
         productions = self.filtered(lambda mo: mo.state not in ['cancel', 'draft', 'done'])
+        productions.components_availability = _('Available')
         for production in productions:
-            forecast_data = []
-            for move in production.move_raw_ids:
-                forecast_data.append(json.loads(move.json_forecast))
-            production.components_availability = 'Available'
-            forecast_data.sort(key=lambda line: (
-                line.get('sortingDate', ''),
-            ), reverse=True)
-            if forecast_data[0].get('sortingDate', False):
-                production.components_availability = _('Exp %s', forecast_data[0]['expectedDate'])
-                production.components_availability_state = 'late' if forecast_data[0]['isLate'] else 'expected'
-            elif any(not data.get('reservedAvailability', False) for data in forecast_data):
-                production.components_availability = 'Not Available'
+            forecast_date = max(production.move_raw_ids.filtered('forecast_expected_date').mapped('forecast_expected_date'), default=False)
+            if forecast_date:
+                production.components_availability = _('Exp %s', format_date(self.env, forecast_date))
+                production.components_availability_state = 'late' if forecast_date > production.date_planned_start else 'expected'
+            elif any(not avai for avai in production.move_raw_ids.mapped('forecast_availability')):
```

Comment:
Maybe it's make more sense to inverse the `if/elif` to just write '*Not Available*' when at least one product isn't available nor expected. Need to be dicuss with our `undefined` PO :suspect:

Reply:
Agree with you, there is no sense to have "Exp <date>" if not all component will be fully available (I wanted to keep old logic). I made the change.